types/react/index.d.ts

```diff
@@ -2774,11 +2775,11 @@ type MergePropTypes<P, T> =
         // If declared props have indexed properties, ignore inferred props entirely as keyof gets widened
         string extends keyof P ? P :
             // Prefer declared types which are not exactly any
-            & Pick<P, NotExactlyAnyPropertyKeys<P>>
+            & DistributivePick<P, NotExactlyAnyPropertyKeys<P>>
```

Comment:
You might not need to have distributive pick peppered over if you just wrap the entire `MergePropTypes` type with a conditional:
```ts
type MergePropTypes<P, T> = P extends any ? /* ... the rest of the stuff */ : never
```

Reply:
Hmm, I'll give it a try, but I think we would have to do `P extends any ? T extends any ? /* ... */ : never : never` since we `Pick` from both `P` and `T`. If that is the case, I feel like three `DistributivePick`s is easier to read than two ternaries.

Reply:
Ah you’re right. I thought we were only distributing against P. I do wonder though if T will ever result in a union since it’s only ever applying against the type of `propTypes`

Reply:
Good point. I suppose it's technically possible for `T` to be a union, but probably not worth supporting. I'll update it and confirm that works.